barretenberg/cpp/src/CMakeLists.txt

```diff
@@ -107,6 +107,7 @@ set(BARRETENBERG_TARGET_OBJECTS
     $<TARGET_OBJECTS:crypto_keccak_objects>
     $<TARGET_OBJECTS:crypto_pedersen_commitment_objects>
     $<TARGET_OBJECTS:crypto_pedersen_hash_objects>
+    $<TARGET_OBJECTS:crypto_poseidon2_objects>
```

Comment:
not sure what this is doing but this helped me resolve some annoying linker errors, presumably introduced because goblin was introduced to main

Reply:
this is necessary from building without it/with it since transcript uses it.